scripts/classification/imagenet/train_imagenet.py

```diff
@@ -281,17 +301,33 @@ def batch_fn(batch, ctx):
 
         return train_data, val_data, batch_fn
 
-    if opt.use_rec:
-        train_data, val_data, batch_fn = get_data_rec(opt.rec_train, opt.rec_train_idx,
-                                                    opt.rec_val, opt.rec_val_idx,
-                                                    batch_size, num_workers)
-    else:
-        train_data, val_data, batch_fn = get_data_loader(opt.data_dir, batch_size, num_workers)
-
-    if opt.mixup:
-        train_metric = mx.metric.RMSE()
-    else:
-        train_metric = mx.metric.Accuracy()
+    def get_data_loader(args):
+        if args.data_backend == 'mxnet':
+            return get_data_rec if args.use_rec else get_data_rec_transfomed
+
+        # Check if DALI is installed:
+        if args.data_backend[0:5] == 'dali-':
+            stream = os.popen("pip list | grep dali")
+            output = stream.read()
+            if output == '':
+                # DALI is not installed
+                logger.info('DALI is not installed\nTrying to install DALI version \'%s\'' % dali_version)
+                ret = os.system('pip install --extra-index-url https://developer.download.nvidia.com/compute/redist ' + dali_version)
```

Comment:
using os.system `pip install` is dangerous as it might trigger another pip, e.g., call this in a system with both pip2/pip3 installed, and run this in python3 environment will trigger pip2 to install dali 

It's possible to import pip like `import pip` and use some of its high level API to install packages, but the current session won't pick it up, again it's not a recommended way.

Reply:
Do you want me to use `pip install` in the bash file AND replace 
```
     ret = os.system('pip install --extra-index-url https://developer.download.nvidia.com/compute/redist ' + dali_version)
```
by warnings?

Reply:
Done. I kept the option to install DALI by 
```
 ret = os.system('pip install ...)
```
but it will be done only when the user will accept this option. Otherwise, (s)he will see the suggestion to install DALI and the corresponding command.